cmd/serverless/api_key.go

```diff
@@ -161,3 +162,17 @@ func hasApiKey() bool {
 		len(os.Getenv(kmsAPIKeyEnvVar)) > 0 ||
 		len(os.Getenv(secretsManagerAPIKeyEnvVar)) > 0
 }
+
+func sendAPIKeyToShell(apiKey string) bool {
+	conn, err := net.Dial("tcp", buildShellHostPort())
+	if err != nil {
+		return false
+	}
+	n, err := conn.Write([]byte(apiKey))
+	return err == nil && n == len(apiKey)
+}
```

Comment:
Why not return an actual error here ?
It would be more idiomatic and more helpful to know what happens, as well as to debug.